docs/lang/articles/reference.md

```diff
@@ -354,14 +362,70 @@ is evaluated to the value of that expression.
 
 ## Simple statements
 
+This section explains the syntax and semantics of compound statements in Taichi. A simple statement is comprised within a single logical line. Several simple statements may occur on a single line separated by semicolons.
+
+```
+simple_stmt ::= expression_stmt
+                | assert_stmt
+                | assignment_stmt
+                | augmented_assignment_stmt
+                | annotated_assignment_stmt
+                | pass_stmt
+                | return_stmt
+                | break_stmt
+                | continue_stmt
+```
+
+
 ### Expression statements
 
+```
+expression_stmt    ::= expression_list
+starred_expression ::= expression | (starred_item ",")* [starred_item]
+starred_item       ::= assignment_expression | "*" or_expr
+```
+
+An expression statement evaluates the expression list (which may be a single expression).
+
 ### Assignment statements
 
+```
+assignment_stmt ::= (target_list "=")+ (expression_list)
+target_list     ::= target ("," target)* [","]
+target          ::= identifier
+                    | "(" [target_list] ")"
+                    | "[" [target_list] "]"
+                    | attributeref
+                    | subscription
+                    | "*" target
+```
+
+An assignment statement evaluates the expression list (remember that this can be a single expression or a comma-separated list,
+the latter yielding a tuple) and assigns the single resulting object to each of the target lists, from left to right.
+
+If the resulting object is a Python value, it will become a Taichi value implicitly by this statement.
```

Comment:
```suggestion
The recursive definition of an assignment statement basically follows
[Python](https://docs.python.org/3/reference/simple_stmts.html#assignment-statements),
with the following points to notice:
- According to the [Variables and scope](#variables-and-scope) section, if a
target is an identifier appearing for the first time, a variable is defined
with that name and inferred type from the corresponding right-hand side
expression. If the expression is evaluated to a Python value, it will be turned
into a Taichi value with [default type](basic/type.md#default-primitive-types-for-integers-and-floating-point-numbers).
- If a target is an existing identifier, the corresponding right-hand side
expression must be evaluated to a Taichi value with the type of the
corresponding variable of that identifier. Otherwise, an implicit cast will
happen.
```